client/src/pages/AddFood.js

```diff
@@ -21,59 +20,76 @@ function AddFood() {
     })
     .required();
   const { register, handleSubmit, reset } = useForm({
-    resolver: yupResolver(schema),
+    resolver: null,
   });
 
   const url = "http://localhost:5000/food";
 
   const mutation = useMutation({
-    mutationFn: (newFood) => {
-      return axios.post(url, newFood).then((response) => {
-        setFood(response.data);
-      });
+    mutationFn: async (newFood) => {
+      console.log(newFood);
+      const config = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newFood),
+      };
+      const response = await fetch(url, config);
+      if (response.ok) {
+        console.log(response);
+        return await response.json();
+      }
     },
   });
 
-  function handleForm(e) {
-    e.preventDefault();
-    mutation.mutate(url, { title: "title", img: "img", author: "author" });
+  function handleForm(data) {
+    console.log(data);
+    mutation.mutate({
+      title: data.title,
+      img: data.img,
+      author: data.author,
+    });
   }
 
-  function handleChange(e) {
-    setFood(e.target.value);
-    console.log(e);
-  }
+  useEffect(() => {
+    reset();
+  }, []);
 
   return (
     <div className="form">
       <Layout />
-      <Form onSubmit={handleSubmit((data) => setData(JSON.stringify(data)))}>
-        <label>Food Name</label>
+      <Form onSubmit={handleSubmit((data) => handleForm(data))}>
+        <label>Food Title</label>
+        <input
+          type="text"
+          name="title"
+          {...register("title")}
+          value={title}
+          onChange={(e) => setTitle(e.target.value)}
```

Comment:
With the use of `register` now in place, a lot of this can be torn down.  Same with the other two fields.

```suggestion
          {...register("title")}
```

Reply:
done